conda_oci_mirror/oci.py

```diff
@@ -104,3 +109,74 @@ def get_info(self, package, tag):
     def get_index_json(self, package, tag):
         digest = self._find_digest(package, tag, C.info_index_media_type)
         return self.get_blob(package, digest).json()
+
+    def push_image(self, _base_path,remote_location, package, _reference, description, _layers):
+
+        manifest_dict = {"schemaVersion":2,"mediaType": "application/vnd.oci.image.manifest.v1+json","config":{}, "layers":[],"annotations":{}}    
+        gh_session = self.oci_auth(package, scope="push,pull")
+        pkg_name = package
+        
+        for layer in _layers:
+
+            r = gh_session.post(f"https://ghcr.io/v2/{self.user_or_org}/{remote_location}/{pkg_name}/blobs/uploads/")
+            headers = r.headers
+            location = headers['location']
+            layer_path = _base_path / layer.file
+
+            #update the manifest
+            _media_type = layer.media_type
+            _size = pathlib.Path(layer_path).stat().st_size
+            digest = sha256sum(layer_path)
+            _digest = "sha256:" + digest
+            
+            annotations_dict = layer.annotations
+            if "empty" not in annotations_dict.keys():
+                #annotations_dict = {"org.conda.md5": _hash_value}
+                infos = {"mediaType":_media_type,"size":_size,"digest":_digest, "annotations": annotations_dict}
+            else:
+                infos = {"mediaType":_media_type,"size":_size,"digest":_digest}
+            manifest_dict["layers"].append(infos)
+
+            # push the layer
+            push_url = f"https://ghcr.io{location}?digest={_digest}"
+            _headers = { "Content-Length": str(_size),"Content-Type": "application/octet-stream"}
+            with open(str(layer_path), "rb") as f:
+                r2 = gh_session.put(push_url, data=f, headers=_headers)
+                print ("response for the currrent layer")
+                print(r2)
+                print(r2.content)
+
+        manifest_dict["annotations"]["org.opencontainers.image.description"] = description
```

Comment:
do not hard code these annotations here.

Reply:
done